File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res = [0] * n
        st = []
        l = r = 0
        while (r < n):
            while st and temperatures[st[-1]] < temperatures[r]:
                res[st[-1]] = r - st[-1]
                st.pop()
            st.append(r)
            r+=1
        return res

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.cpp
class Solution {
public:
    int calculator(int x, int y, char operand) {
        switch(operand) {
            case '+': return x+y; break;
            case '-': return x-y; break;
            case '*': return x*y; break;
            case '/': return x/y; break;
        }
        return 0;
    }
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        unordered_set<string> operands = {"+", "-", "*", "/"};
        for(string& s : tokens) {
            if(operands.find(s) != operands.end()) {
                int b = st.top(); st.pop();
                int a = st.top(); st.pop();

                 // Sending char instead of string with s[0] because switch does not support it
                int res = calculator(a, b , s[0]);
                st.push(res);
            } else {
                st.push(stoi(s));
            }
        }
        return st.top();
    }
};
